### PR TITLE
op-guide: add tidb-config-file.md

### PR DESCRIPTION
**Pull request by winkyao**

### CODE REVIEW

**Comment by winkyao** [comment]
@shenli @coocood @lilin90 PTAL

File: op-guide/tidb-config-file.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+## performance
+
+性能相关配置。
+
+### `max-procs`
+
++ TiDB 的 CPU 使用数量。
++ 默认: 0
++ 默认为 0 表示使用机器上所有的 CPU，也可以设置成 `max-procs`， 那么 TiDB 会使用 `max-procs` 个 CPU 数量。
```
Comment (coocood) [inline op-guide/tidb-config-file.md]: s/max-procs/n

File: op-guide/tidb-config-file.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
+---
+title: TiDB 配置文件解释
+category: deployment
+---
+
+# TiDB 配置文件
+
+TiDB 配置文件比命令行参数支持更多的选项。大家可以在 [config/config.toml.example](https://github.com/pingcap/tidb/blob/master/config/config.toml.example) 找到默认的配置文件。大家重命名为 `config.toml` 即可。
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: -> 你可以在 [config/config.toml.example](https://github.com/pingcap/tidb/blob/master/config/config.toml.example) 找到默认的配置文件，重命名为 `config.toml` 即可。

File: op-guide/tidb-config-file.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+这里只阐述在命令行参数中没有的参数，命令行参数大家可以看[这里](configuration.md)
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: -> 本文档只阐述未包含在命令行参数中的参数，命令行参数参见[这里](configuration.md)。

File: op-guide/tidb-config-file.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+### `split-table`
+
++ 为每个 table 建立单独的 region。
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: region - Region

File: op-guide/tidb-config-file.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
++ 默认: true
++ 如果需要创建大量的表，我们建议把这个参数设置为 false。
+
+### `oom-action`
+
++ 指定 TiDB 发生 out-of-memory 错误时的操作。
++ 默认: "log"
++ 现在合法的选项是 ["log", "cancel"]，如果为 "log", 仅仅是打印日志，不作实质处理。如果为 "cancel"，我们会取消执行这个操作，并且输出日志。
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: "," before 仅仅 -> "，"

File: op-guide/tidb-config-file.md [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+
++ 日志最大保留的天数。
++ 默认: 0
++ 默认是不清理的，如果设置了，在 `max-days` 之后 TiDB 会清理过期的日志文件。
+
+#### `max-backups`
+
++ 保留的日志的最大数量。
++ 默认: 0
++ 默认全部保存，如果设置为7，会最多保留 7 个老的日志文件。
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: Leave a whitespace before the first 7.

File: op-guide/tidb-config-file.md [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+
++ TiDB 的 CPU 使用数量。
++ 默认: 0
++ 默认为 0 表示使用机器上所有的 CPU，也可以设置成 `max-procs`， 那么 TiDB 会使用 `max-procs` 个 CPU 数量。
+
+### `stmt-count-limit`
+
++ TiDB 一个事务允许的最大语句条数限制。
++ 默认: 5000
++ 在一个事务中，超过 `stmt-count-limit` 条语句后还没有 roolback 或者 commit，TiDB 将会返回 `statement count 5001 exceeds the transaction limitation, autocommit = false` 错误。
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: roolback -> rollback

File: op-guide/tidb-config-file.md [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+
+### `tcp-keep-alive`
+
++ TiDB 在 tcp 层开启 keepalive
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: tcp ->  TCP

File: op-guide/tidb-config-file.md [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
++ 看数据量和数据分布情况，一般情况下是越多越好，数值越大对 CPU 开销越大
+
+### `cross-join`
+
++ 默认: true
++ 在做 join 的时候，两边表没有任何条件（where 字段），默认可以执行这样的语句。但是设置为 false，则如有这样的 join 语句出现，server 会拒绝执行
+
+### `stats-lease`
+
++ TiDB 统计信息做 Analyze 和 重载统计信息的时间间隔。
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: Delete the extra whitespace after "和".

File: op-guide/tidb-config-file.md [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
++ 默认: xxx
++ xxxx
+
+### `run-auto-analyze`
+
+TiDB 是否做自动的 Analyze。
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: Add a bullet list sign "+" for this line.

File: op-guide/tidb-config-file.md [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+### `grpc-connection-count`
+
++ 跟每个 tikv 之间建立的最大连接数。
++ 默认: 16
+
+### `commit-timeout`
+
++ 执行事务提交时，最大的超时时间。
++ 默认: 41s
++ 这个值必须是大于两倍 raft 选举的超时时间。
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: raft -> Raft

File: op-guide/tidb-config-file.md [hunk 12]
```diff
@@ -0,0 +1,224 @@
...
+### `capacity`
+
++ 缓存语句的数量。
++ 默认: 100
+
+## tikv-client
+
+### `grpc-connection-count`
+
++ 跟每个 tikv 之间建立的最大连接数。
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: tikv -> TiKV

File: op-guide/tidb-config-file.md [hunk 13]
```diff
@@ -0,0 +1,224 @@
+---
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: Please add it to README.

File: op-guide/tidb-config-file.md (hunk 1 above)
Comment (winkyao) [reply op-guide/tidb-config-file.md]: s/max-procs/n/g?

File: op-guide/tidb-config-file.md (hunk 13 above)
Comment (winkyao) [reply op-guide/tidb-config-file.md]: Done

**Comment by winkyao** [comment]
@lilin90 @coocood PTAL

**Comment by coocood** [comment]
LGTM

File: op-guide/tidb-config-file.md [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
++ 缓存语句的数量。
++ 默认: 2560
+
+### `shards`
+
++ plan-cache 桶的数量。
++ 默认: 256
++ 这个数量越大，锁的粒度越小。
+
+### prepared-plan-cache
```
Comment (lilin90) [inline op-guide/tidb-config-file.md]: Is this title the same level as `## plan-cache`?

File: op-guide/tidb-config-file.md [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
++ 默认: 10000
++ 当查询的行数(包括中间结果，基于统计信息)大于这个值，我们就会当成是一个 `expensive` 的操作，输出一个前缀带有 `[EXPENSIVE_QUERY]` 的日志。
+
+### `query-log-max-len`
+
++ 最长的 SQL 输出长度。
++ 默认: 2048
++ 当语句的长度大于 `query-log-max-len`，将会被截断输出。
+
+### log.file
```
Comment (XuHuaiyu) [inline op-guide/tidb-config-file.md]: useless?

File: op-guide/tidb-config-file.md [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+
+### `stats-lease`
+
++ TiDB 重载统计信息, 更新表行数, 检查是否需要自动 analyze 以及加载列的统计信息的时间间隔
++ 默认： 3s
+    - 每隔 `stats-lease` 时间， TiDB 会检查统计信息是否有更新，如果有会将其更新到内存中
+    - 每隔 `5 * stats-lease` 时间，TiDB 会将 DML 产生的总行数以及修改的行数变化持久化下来
+    - 每隔 `stats-lease` 时间，TiDB 会检查是否有表或者索引需要自动 analyze
+    - 每隔 `stats-lease` 时间，TiDB 会检查是否有列的统计信息需要被加载到内存中
+
```
Comment (XuHuaiyu) [inline op-guide/tidb-config-file.md]: useless line

File: op-guide/tidb-config-file.md [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+### `cross-join`
+
++ 默认: true
++ 在做 join 的时候，两边表没有任何条件（where 字段），默认可以执行这样的语句。但是设置为 false，则如有这样的 join 语句出现，server 会拒绝执行
+
+### `stats-lease`
+
++ TiDB 重载统计信息, 更新表行数, 检查是否需要自动 analyze 以及加载列的统计信息的时间间隔
++ 默认： 3s
```
Comment (XuHuaiyu) [inline op-guide/tidb-config-file.md]: Style of `:` is not consistent with the others.

File: op-guide/tidb-config-file.md [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+
+### `run-auto-analyze`
+
++ TiDB 是否做自动的 Analyze。
++ 默认: true
+
+### `feedback-probability`
+
++ TiDB 对查询收集统计信息反馈的概率
++ 默认： 0
```
Comment (XuHuaiyu) [inline op-guide/tidb-config-file.md]: ditto

File: op-guide/tidb-config-file.md (hunk 15 above)
Comment (winkyao) [reply op-guide/tidb-config-file.md]: log.file is a new level.

**Comment by winkyao** [comment]
@XuHuaiyu PTAL

File: op-guide/tidb-config-file.md [hunk 19]
```diff
@@ -0,0 +1,228 @@
...
++ TiDB 在 TCP 层开启 keepalive 
++ 默认: false
+
+### `retry-limit`
+
++ TiDB 在提交事务的时候遇到 key 冲突或者其他错误时进行的重试次数。
++ 默认: 10
++ 如果 TiDB 超过 `retry-limit` 次重试还是没有成功，将会返回错误。
+
+### `join-concurrency`
```
Comment (XuHuaiyu) [inline op-guide/tidb-config-file.md]: This is removed in https://github.com/pingcap/tidb/pull/6244

**Pull request merged**